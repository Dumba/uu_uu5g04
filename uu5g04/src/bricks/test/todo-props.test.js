/**
 * Copyright (C) 2019 Unicorn a.s.
 * 
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 * 
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import React from 'react';
import {shallow} from 'enzyme';
import UU5 from "uu5g04";
import "uu5g04-bricks";
import enzymeToJson from 'enzyme-to-json';
import TestTools from "../../core/test/test-tools.js";


const TagName = "UU5.Bricks.Todo";

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.SectionMixin",
    "UU5.Common.ContentMixin",
    "UU5.Common.LevelMixin",
    "UU5.Common.NestingLevelMixin"
  ],
  props: {
    props: {
      values: [{ key: 'value', num: 3, bool: true }]
    }
  },
  requiredProps: {},
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false
    },
    enzymeToJson: true
  }
};

describe(`${TagName}`, () => {
  TestTools.testProperties(TagName, CONFIG);
});


describe(`${TagName} docKit examples`, () => {

  it(`${TagName} should render without crash`, () => {
    const wrapper = shallow(
      <UU5.Bricks.Todo id={"uuID"} props={{ key: 'value', num: 3, bool: true }} />
    );
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

});










