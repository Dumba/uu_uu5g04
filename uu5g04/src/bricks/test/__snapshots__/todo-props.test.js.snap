// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Todo checkGrammar=false 1`] = `
<Section
  checkGrammar={false}
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo checkGrammar=true 1`] = `
<Section
  checkGrammar={true}
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo checkHighlight=false 1`] = `
<Section
  checkHighlight={false}
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo checkHighlight=true 1`] = `
<Section
  checkHighlight={true}
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo checkSpaces=false 1`] = `
<Section
  checkSpaces={false}
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo checkSpaces=true 1`] = `
<Section
  checkSpaces={true}
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo className="my-classname" 1`] = `
<Section
  className="uu5-bricks-todo my-classname"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo content="This is Content from ContentMixin" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo content=12345 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo controlled=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo controlled=true 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo default props 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo disabled=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo disabled=true 1`] = `
<Section
  className="uu5-bricks-todo"
  disabled={true}
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo docKit examples UU5.Bricks.Todo should render without crash 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre>
    {
  "key": "value",
  "num": 3,
  "bool": true
}
  </Pre>
</Section>
`;

exports[`UU5.Bricks.Todo dynamic=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo dynamic=true 1`] = `
<Section
  className="uu5-bricks-todo"
  dynamic={true}
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo footer={"type":"div","key":null,"ref":null,"props":{"children":"Footer of section"},"_owner":null,"_store":{}} 1`] = `
<Section
  className="uu5-bricks-todo"
  footer={
    <div>
      Footer of section
    </div>
  }
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo header={"type":"div","key":null,"ref":null,"props":{"children":"Header of section"},"_owner":null,"_store":{}} 1`] = `
<Section
  className="uu5-bricks-todo"
  header={
    <div>
      Header of section
    </div>
  }
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo hidden=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo hidden=true 1`] = `
<Section
  className="uu5-bricks-todo"
  hidden={true}
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo id="myId" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="myId-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo ignoreHTML=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo ignoreHTML=true 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level="0" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level="0"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level="1" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level="1"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level="2" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level="2"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level="3" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level="3"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level="4" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level="4"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level="5" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level="5"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level="6" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level="6"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level=0 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level={0}
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level=1 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level={1}
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level=2 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level={2}
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level=3 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level={3}
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level=4 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level={4}
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level=5 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level={5}
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo level=6 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  level={6}
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo mainAttrs=undefined 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo name="myName" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  name="myName-inner"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo nestingLevel="bigBox" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBox"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo nestingLevel="bigBoxCollection" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo nestingLevel="box" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="box"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo nestingLevel="boxCollection" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="boxCollection"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo nestingLevel="container" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="container"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo nestingLevel="inline" 1`] = `""`;

exports[`UU5.Bricks.Todo nestingLevel="page" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="page"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo nestingLevel="smallBox" 1`] = `""`;

exports[`UU5.Bricks.Todo nestingLevel="smallBoxCollection" 1`] = `""`;

exports[`UU5.Bricks.Todo nestingLevel="spa" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="spa"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo noIndex=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo noIndex=true 1`] = `
<Section
  className="uu5-bricks-todo uu5-noindex"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo props={"key":"value","num":3,"bool":true} 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre>
    {
  "key": "value",
  "num": 3,
  "bool": true
}
  </Pre>
</Section>
`;

exports[`UU5.Bricks.Todo selected=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo selected=true 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  selected={true}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo style="padding:8px; borderRadius: 2px" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo style={"backgroundColor":"red","color":"blue"} 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo textCorrector=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  textCorrector={false}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo textCorrector=true 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
  textCorrector={true}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo tooltip="My Tooltip" 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={
    Object {
      "title": "My Tooltip",
    }
  }
  tooltip="My Tooltip"
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo underline=false 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;

exports[`UU5.Bricks.Todo underline=true 1`] = `
<Section
  className="uu5-bricks-todo"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <Pre />
</Section>
`;
